### PR TITLE
Add basic test_package docs

### PR DESCRIPTION
**Pull request by czoido**
Goes with: https://github.com/conan-io/examples2/pull/54

### CODE REVIEW

File: tutorial/creating_packages/test_conan_packages.rst [hunk 1]
```diff
@@ -1,2 +1,171 @@
...
+    In this example, we retrieve the *fmt* and *gtest* Conan packages from a Conan
+    repository with packages compatible with Conan 2.0. To run this example successfully
+    you should add this remote to your Conan configuration (if did not already do it)
+    doing: ``conan remote add conanv2
+    https://conanv2beta.jfrog.io/artifactory/api/conan/conan --index 0``
+
+
+In all the previous sections of the tutorial, we used the *test_package*. It was invoked
+automatically at the end of the ``conan create`` command after building our package
+verifying that the package is created correctly. Let’s explain the *test_package* in more
```
Comment (RubenRBS) [inline tutorial/creating_packages/test_conan_packages.rst:15]:
```suggestion
verifying that the package is created correctly. Let's explain the *test_package* in more
```

File: tutorial/creating_packages/test_conan_packages.rst [hunk 2]
```diff
@@ -1,2 +1,171 @@
-Test Conan packages
-===================
+Testing Conan packages: the test() method
```
Comment (memsharded) [inline tutorial/creating_packages/test_conan_packages.rst:1]: This section is not only about ``test()`` method, but all ``test_package``? Maybe drop the ``test()`` from the title?

File: tutorial/creating_packages/test_conan_packages.rst [hunk 3]
```diff
@@ -1,2 +1,171 @@
...
+        def build(self):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+
+        def layout(self):
+            cmake_layout(self)
+
+        def test(self):
+            if not cross_building(self):
```
Comment (memsharded) [inline tutorial/creating_packages/test_conan_packages.rst:104]: Maybe use better ``can_run()`` here? There are things that are cross-build, but still can run, like Mac with different architectures using emulation?

File: tutorial/creating_packages/test_conan_packages.rst [hunk 4]
```diff
@@ -1,2 +1,171 @@
...
+  consumers will be able to link against it and reuse it.
+
+* It is a small Conan project itself, it contains its own *conanfile.py*, and its source
+  code including build scripts, that depends on the package being created, and builds and
+  execute a small application that requires the library in the package.
+
+* It doesn’t belong to the package. It only exist in the source repository, not in the
+  package.
+
+The *test_package* folder for for our hello/1.0 Conan package has the following contents:
```
Comment (RubenRBS) [inline tutorial/creating_packages/test_conan_packages.rst:40]:
```suggestion
The *test_package* folder for our hello/1.0 Conan package has the following contents:
```

File: tutorial/creating_packages/test_conan_packages.rst [hunk 5]
```diff
@@ -1,2 +1,171 @@
...
+
+.. code-block:: text
+
+   test_package
+    ├── CMakeLists.txt
+    ├── conanfile.py
+    └── src
+        └── example.cpp
+
+Let's have a look at the different files that are part of the *test_package*. First, the
```
Comment (RubenRBS) [inline tutorial/creating_packages/test_conan_packages.rst:50]:
```suggestion
Let's have a look at the different files that are part of the *test_package*. First,
```

File: tutorial/creating_packages/test_conan_packages.rst [hunk 6]
```diff
@@ -1,2 +1,171 @@
...
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+
+        def layout(self):
+            cmake_layout(self)
+
+        def test(self):
+            if not cross_building(self):
+                cmd = os.path.join(self.cpp.build.bindirs[0], "example")
```
Comment (memsharded) [inline tutorial/creating_packages/test_conan_packages.rst:105]: We accept now ``os.path.join(self.cpp.build.bindir, "example")``
```suggestion
                cmd = os.path.join(self.cpp.build.bindir, "example")
```

File: tutorial/creating_packages/test_conan_packages.rst [hunk 7]
```diff
@@ -1,2 +1,171 @@
...
+.. code-block:: cpp
+    :caption: *test_package/src/example.cpp*
+
+    #include "hello.h"
+
+    int main() {
+        hello();
+    }
+
+Then the *CMakeLists.txt* to tell CMake how to build the example:
```
Comment (RubenRBS) [inline tutorial/creating_packages/test_conan_packages.rst:63]:
```suggestion
Then the *CMakeLists.txt* file to tell CMake how to build the example:
```

**Review by memsharded — APPROVED** [review]

File: tutorial/creating_packages/test_conan_packages.rst [hunk 8]
```diff
@@ -1,2 +1,171 @@
...
+            cmake_layout(self)
+
+        def test(self):
+            if not cross_building(self):
+                cmd = os.path.join(self.cpp.build.bindirs[0], "example")
+                self.run(cmd, env="conanrun")
+
+Let's go through the most relevant parts:
+
+* We add the requirements in the ``requirements()`` method but in this case, we use the
```
Comment (RubenRBS) [inline tutorial/creating_packages/test_conan_packages.rst:110]:
```suggestion
* We add the requirements in the ``requirements()`` method, but in this case we use the
```

File: tutorial/creating_packages/test_conan_packages.rst [hunk 9]
```diff
@@ -1,2 +1,171 @@
...
+    [ 50%] Building CXX object CMakeFiles/example.dir/src/example.cpp.o
+    [100%] Linking CXX executable example
+    [100%] Built target example
+
+    -------- Testing the package: Running test() --------
+    hello/1.0 (test package): Running test()
+    hello/1.0 (test package): RUN: ./example
+    hello/1.0: Hello World Release! (with color!)
+
+As you can see in the output, our test_package builds successfully testing that the
```
Comment (RubenRBS) [inline tutorial/creating_packages/test_conan_packages.rst:163]:
```suggestion
As you can see in the output, our *test_package* builds successfully testing that the
```
It's how the rest of the file refers to it

**Review by RubenRBS — APPROVED** [review]
😃

File: tutorial/creating_packages/test_conan_packages.rst (hunk 6 above)
Comment (czoido) [reply tutorial/creating_packages/test_conan_packages.rst:105]: Maybe we should also modify the 2.0 templates to use `bindir` and `can_run` in the test_package

Comment (memsharded) [reply tutorial/creating_packages/test_conan_packages.rst:105]: Definitely, I'll do it right now.

Comment (memsharded) [reply tutorial/creating_packages/test_conan_packages.rst:105]: https://github.com/conan-io/conan/pull/12765

**Pull request merged**